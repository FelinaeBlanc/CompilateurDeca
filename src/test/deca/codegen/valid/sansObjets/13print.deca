// Description:
//    Test les differents print et print hex
//
// Resultats:
//    
//
// Historique:
//    cree le @DATE@

{
    int x = 5;
    float a = 10.5;


    // test print simple int
    println("print int: ", x);

    // test print simple float
    println("print float: ", a);

    // test print hex float
    printlnx("print hex float: ", a);

    // test print float et int (+ string)
    println("print float et int: ", a, x);

    // test print expressions (int)
    println("print expr type int: ", (x + 15) / 2);

    // test print expressions (float)
    print("print expr type float: ", (a + 10) / 2.0);

}
