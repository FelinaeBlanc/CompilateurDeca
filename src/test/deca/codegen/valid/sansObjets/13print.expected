print int: 5
print float: 1.05000e+01
print hex float: 0x1.5p+3
print float et int: 1.05000e+015
print expr type int: 10
print expr type float: 1.02500e+01
